needle-notion-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,154 @@
+import type {
+  ConnectorRequest,
+  CreateConnectorRequest,
+} from "~/models/connectors-models";
+import {
+  createConnector,
+  getConnector,
+  listConnectors,
+  deleteConnector,
+  publishConnectorRun,
+  createNeedleFileId,
+  type Session,
+  type ConnectorRunDescriptor,
+} from "@needle-ai/needle-sdk";
+import { db } from "../db";
+import { eq } from "drizzle-orm";
+import { notionConnectorsTable, notionPagesTable } from "../db/schema";
+import { Client as NotionClient } from "@notionhq/client";
+import type {
+  DatabaseObjectResponse,
+  PageObjectResponse,
+} from "@notionhq/client/build/src/api-endpoints";
+
+export async function createNotionConnector(
+  request: CreateConnectorRequest,
+  session: Session,
+) {
+  const connector = await createConnector(
+    {
+      name: request.name,
+      cronJob: request.cronJob,
+      cronJobTimezone: request.cronJobTimezone,
+      collectionIds: [request.collectionId],
+      credentials: request.notionToken.access_token,
+    },
+    session.id,
+  );
+
+  const pagesToInsert = request.notionPages.map((p) => ({
```

Comment:
I am more fan of DRY. I mean that in the function below (`runNotionConnector`) we anyways fetch the pages to track. So that function takes care of fetching + updating DB for notion pages. Let's do it only there? We can skip this `insert` and dont have to pass pages down to here from the UI.